server/models/ride.ts

```diff
@@ -7,32 +10,31 @@ export enum Type {
 }
 
 export type RideType = {
-  type: Type,
   id: string,
-  startLocation: string,
-  endLocation: string,
+  type: Type,
+  startLocation: LocationType,
+  endLocation: LocationType,
   startTime: string,
   endTime: string,
-  riderId: string,
-  driverId?: string,
+  rider: RiderType,
+  driver?: DriverType,
 };
 
 const schema = new dynamoose.Schema({
-  type: {
+  id: {
     hashKey: true,
     type: String,
-    enum: Object.values(Type),
   },
-  id: {
-    rangeKey: true,
+  type: {
     type: String,
+    enum: Object.values(Type),
   },
-  startLocation: String,
-  endLocation: String,
+  startLocation: Location as any,
+  endLocation: Location as any,
```

Comment:
So if someone inputted a random address into a text field, we'd have to store it as a Location in the DB? Seems like that could get out of hand. Let me know if I am misunderstanding anything. We'll definitely have to consider this, but it could just be fixed in a future PR.